validator/client/src/main/java/tech/pegasys/teku/validator/client/ValidatorClientService.java

```diff
@@ -114,7 +114,11 @@ public static ValidatorClientService create(
     Optional<RestApi> validatorRestApi = Optional.empty();
     if (validatorApiConfig.isRestApiEnabled()) {
       validatorRestApi =
-          Optional.of(ValidatorRestApi.create(validatorApiConfig, new KeyManager(validatorLoader)));
+          Optional.of(
+              ValidatorRestApi.create(
+                  validatorApiConfig,
+                  new KeyManager(
+                      validatorLoader, services.getDataDirLayout().getValidatorDataDirectory())));
```

Comment:
if we can get the ownedValidators list to maintain readonly, KeyManager probably (hopefully) won't even need to know about the validator data directory, unless we need the actual directory to write a key to I guess, but currently with list, we shouldn't need it...